pantheon/src/main/java/tech/pegasys/pantheon/Runner.java

```diff
@@ -195,7 +196,8 @@ private void writePantheonPortsToFile() {
     }
   }
 
-  public Optional<? extends Peer> getAdvertisedPeer() {
-    return networkRunner.getNetwork().getAdvertisedPeer();
+  @VisibleForTesting
+  P2PNetwork getP2PNetwork() {
```

Comment:
It's quite unfortunate to have to expose the whole `P2PNetwork`.  Is it possible for this to only return the local enode (via delegation)?